syntax = "proto3";

package com.empuje.eventservice.grpc.gen;

import "google/protobuf/timestamp.proto";

option java_multiple_files = true;
option java_package = "com.empuje.eventservice.grpc.gen";
option java_outer_classname = "EventServiceProto";

service EventService {
  rpc CreateEvent(CreateEventRequest) returns (EventResponse) {}
  rpc UpdateEvent(UpdateEventRequest) returns (EventResponse) {}
  rpc DeleteEvent(DeleteEventRequest) returns (DeleteEventResponse) {}
  rpc ListEvents(ListEventsRequest) returns (stream EventResponse) {}
  rpc AssignMember(AssignMemberRequest) returns (EventResponse) {}
  rpc RemoveMember(RemoveMemberRequest) returns (EventResponse) {}
}

message CreateEventRequest {
  string name = 1;
  string description = 2;
  google.protobuf.Timestamp event_datetime = 3;
  repeated int64 participant_ids = 4;
  int64 created_by = 5;
}

message UpdateEventRequest {
  int64 id = 1;
  string name = 2;
  string description = 3;
  google.protobuf.Timestamp event_datetime = 4;
  repeated int64 participant_ids = 5;
  int64 updated_by = 6;
}

message EventResponse {
  int64 id = 1;
  string name = 2;
  string description = 3;
  google.protobuf.Timestamp event_datetime = 4;
  repeated int64 participant_ids = 5;
  google.protobuf.Timestamp created_at = 6;
}

message ListEventsRequest {}

message DeleteEventRequest { int64 id = 1; int64 requested_by = 2; }
message DeleteEventResponse { bool success = 1; string message = 2; }

message AssignMemberRequest { int64 event_id = 1; int64 user_id = 2; int64 assigned_by = 3; }
message RemoveMemberRequest { int64 event_id = 1; int64 user_id = 2; int64 removed_by = 3; }
